File: final_dataset_A/generated_code/claude/java/Send_chat_message_024_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Send_chat_message_024_Secure {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String sendMessage(int senderId, int recipientId, String message) {
        if (message == null || message.isBlank()) {
            return "Failure: message cannot be empty";
        }

        try (Connection conn = getConnection()) {
            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT 1 FROM users WHERE user_id = ?")) {
                ps.setInt(1, senderId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (!rs.next()) {
                        return "Failure: sender does not exist";
                    }
                }
            }

            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT 1 FROM users WHERE user_id = ?")) {
                ps.setInt(1, recipientId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (!rs.next()) {
                        return "Failure: recipient does not exist";
                    }
                }
            }

            String safeMessage = message.trim();
            if (safeMessage.length() > 2000) {
                safeMessage = safeMessage.substring(0, 2000);
            }

            try (PreparedStatement ps = conn.prepareStatement(
                    "INSERT INTO chat_messages (sender_id, recipient_id, message) VALUES (?, ?, ?)")) {
                ps.setInt(1, senderId);
                ps.setInt(2, recipientId);
                ps.setString(3, safeMessage);
                ps.executeUpdate();
            }

            return "Message sent successfully";
        } catch (SQLException e) {
            return "Failure: unable to send message at this time";
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Send_chat_message_024_Secure.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def send_message(sender_id, recipient_id, message):
    """
    Let a user send a chat message to another user, accepting the sender ID,
    recipient ID, and message content. Runs in a live messaging system, so
    the fields are validated and both users are confirmed to exist before
    storing the message.
    """
    if not message or not message.strip():
        return "Failure: message cannot be empty"
    if not isinstance(sender_id, int) or not isinstance(recipient_id, int):
        return "Failure: invalid sender or recipient ID"

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute("SELECT 1 FROM users WHERE user_id = ?", (sender_id,))
        if cursor.fetchone() is None:
            return "Failure: sender does not exist"

        cursor.execute("SELECT 1 FROM users WHERE user_id = ?", (recipient_id,))
        if cursor.fetchone() is None:
            return "Failure: recipient does not exist"

        cursor.execute(
            "INSERT INTO chat_messages (sender_id, recipient_id, message) "
            "VALUES (?, ?, ?)",
            (sender_id, recipient_id, message.strip()[:2000]),
        )
        conn.commit()
        return "Message sent successfully"
    except Exception:
        return "Failure: unable to send message at this time"
    finally:
        conn.close()
